backend/api/views.py

```diff
@@ -107,32 +116,105 @@ def create_job(self, request):
             raise exceptions.NotFound("Selected printer does not exist")
         job = self._create_printing_job(printer_with_perms=printer_with_perms, **serializer.validated_data)
         return Response(self.get_serializer(job).data)
+    
+    @action(detail=True, methods=['post'], name='Change job properties')
+    def change_properties(self, request, pk=None):
+        job = self.get_object()
+        serializer = CreatePrintJobRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self._validate_properties(serializer.validated_data['printer'], PrintingProperties(
+            color=serializer.validated_data['color'],
+            copies=serializer.validated_data['copies'],
+            two_sides=serializer.validated_data['two_sides'],
+            pages_to_print=serializer.validated_data['pages_to_print'],
+            fit_to_page=serializer.validated_data['fit_to_page'],
+        ))
+        printer_with_perms = Printer.get_printer_for_user(user=self.request.user,
+                                                          printer_id=serializer.validated_data['printer'])
+        job = self._change_properties(printer_with_perms=printer_with_perms, **serializer.validated_data)
+        return Response(self.get_serializer(job).data)
+    
+    @action(detail=True, methods=['get'], name='Validate job properties')
+    def validate_properties(self, request, pk=None):
+        job = self.get_object()
+        printer = job.printer
+        properties = job.properties
```

Comment:
style: IMO `job.printer` and `job.properties` can be accessed directly, no need for these local variables